src/test/scala/com/mozilla/telemetry/views/MainSummaryViewTest.scala

```diff
@@ -2325,4 +2331,125 @@ class MainSummaryViewTest extends FlatSpec with Matchers with DataFrameSuiteBase
+          "clientId" -> clientId1,
+          "sample_id" -> 42,
+          "document_id" -> docId2,
+          "normalized_channel" -> "release",
+          "normalized_os_version" -> "10.0",
+          "metadata" -> Map(
+            "header" -> Map("date" -> "Mon, 1 Jul 2019 00:00:00 GMT"),
+            "geo" -> Map(
+              "country" -> "US",
+              "city" -> "Hermiston",
+              "subdivision1" -> "OR",
+              "subdivision2" -> "Umatilla"
+            ),
+            "uri" -> Map(
+              "app_update_channel" -> "release"
+            )
+          )
+        ))
+      ),
+      Map(
+        "attributeMap" -> baseAttributes,
+        "payload" -> (basePayload + ("document_id" -> docId3))
+      )
+    ).map(
+      msg => asJson(msg + ("payload" -> base64(asJson(msg("payload")))))
+    ).zipWithIndex.foreach {
+      case (line, index) => writeTextFile(s"$prefix/$index.ndjson.gz", s"$line\n", compression)
+    }
+
+    ndjson.Dataset invokePrivate PrivateMethod[ndjson.Dataset]('useTestingPath)(datasetTestingPath)
+
+    spark.conf.set("mozilla.doNotStopSparkContext", "true")
+
+    MainSummaryView.main(Array(
+      "--from=20190101",
+      "--to=20190101",
+      "--input-source=ndjson",
+      "--output-file-system=file",
+      s"--bucket=$tempDir/output/ndjson"
+    ))
+
+    val result = spark.read.parquet(
+      s"file://$tempDir/output/ndjson/main_summary/v4/submission_date_s3=20190101"
+    ).select(
+      "document_id", "client_id", "sample_id", "channel", "normalized_channel",
+      "normalized_os_version", "country", "city", "geo_subdivision1", "geo_subdivision2",
+      "submission_date", "timestamp", "client_submission_date", "client_clock_skew"
+    ).orderBy("document_id").toJSON.collect.toList
+    val expect = List(
+      s"""{"document_id":"$docId1","client_id":"$clientId1","sample_id":42,"channel":
+         |"metadata.uri.app_update_channel","normalized_channel":"normalized_channel",
+         |"normalized_os_version":"normalized_os_version","country":"metadata.geo.country",
+         |"city":"metadata.geo.city","geo_subdivision1":"metadata.geo.subdivision1",
+         |"geo_subdivision2":"metadata.geo.subdivision2","submission_date":"20190701",
+         |"timestamp":1561939200000000000,"client_submission_date":"metadata.header.date"
+         |}""".replaceAll("\\s+\\|", ""),
+      s"""{"document_id":"$docId2","client_id":"$clientId1","sample_id":42,"channel":"release",
+         |"normalized_channel":"release","normalized_os_version":"10.0","country":"US","city":
+         |"Hermiston","geo_subdivision1":"OR","geo_subdivision2":"Umatilla",
+         |"submission_date":"20190701","timestamp":1561939200000000000,
+         |"client_submission_date":"Mon, 1 Jul 2019 00:00:00 GMT","client_clock_skew":0
+         |}""".replaceAll("\\s+\\|", ""),
+      s"""{"document_id":"$docId3","submission_date":"20190701","timestamp":1561939200000000000}"""
+    )
+    assert(result == expect)
```

Comment:
```suggestion
    result should contain theSameElementsAs expect
```
this will be more robust.

Reply:
i did not know how to do that, TYSM